src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -1630,7 +1630,35 @@ void CodeGen::genCodeForShift(GenTree* tree)
         unsigned immWidth   = emitter::getBitWidth(size); // For ARM64, immWidth will be set to 32 or 64
         unsigned shiftByImm = (unsigned)shiftBy->AsIntCon()->gtIconVal & (immWidth - 1);
 
-        GetEmitter()->emitIns_R_R_I(ins, size, tree->GetRegNum(), operand->GetRegNum(), shiftByImm);
+#ifdef TARGET_ARM64
+        // Check if we can recognize ubfiz/sbfiz idiom in LSH(CAST(X), CNS) pattern
+        if (tree->gtGetOp1()->OperIs(GT_CAST) && tree->gtGetOp1()->isContained())
+        {
+            GenTreeCast* cast  = tree->gtGetOp1()->AsCast();
+            int          width = 0;
+            if (varTypeIsSmall(cast->CastToType()))
+            {
+                // TYP_INT <- %SMALL_(U)INT% <- TYP_INT
+                width = (int)genTypeSize(cast->CastToType()) * 8;
+                assert((width == 16) || (width == 8));
+                assert((cast->CastFromType() == TYP_INT) && cast->TypeIs(TYP_INT));
```

Comment:
```suggestion
                assert(cast->CastFromType() == TYP_INT);
```
Don't think it is necessary to check this, it should be guaranteed by IR.

Reply:
I'd prefer to keep it, e.g. I don't expect TYP_LONG here or TYP_REF - such asserts help when we trash IR/introduce new transformations